extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java

```diff
@@ -1,670 +1,671 @@
+    }
+
+    private abstract static class AbstractCompositeApplicationClassesPredicate<T> implements Predicate<T> {
+
+        private final IndexView applicationClassesIndex;
+        private final Set<DotName> generatedClassNames;
+        private final List<ApplicationClassPredicateBuildItem> applicationClassPredicateBuildItems;
+        private final Optional<TestClassPredicateBuildItem> testClassPredicate;
+
+        protected abstract DotName getDotName(T t);
+
+        private AbstractCompositeApplicationClassesPredicate(IndexView applicationClassesIndex,
+                Set<DotName> generatedClassNames,
+                List<ApplicationClassPredicateBuildItem> applicationClassPredicateBuildItems,
+                Optional<TestClassPredicateBuildItem> testClassPredicate) {
+            this.applicationClassesIndex = applicationClassesIndex;
+            this.generatedClassNames = generatedClassNames;
+            this.applicationClassPredicateBuildItems = applicationClassPredicateBuildItems;
+            this.testClassPredicate = testClassPredicate;
+        }
+
+        @Override
+        public boolean test(T t) {
+            final DotName dotName = getDotName(t);
+            if (applicationClassesIndex.getClassByName(dotName) != null) {
+                return true;
+            }
+            if (generatedClassNames.contains(dotName)) {
+                return true;
+            }
+            String className = dotName.toString();
+            if (!applicationClassPredicateBuildItems.isEmpty()) {
+                for (ApplicationClassPredicateBuildItem predicate : applicationClassPredicateBuildItems) {
+                    if (predicate.test(className)) {
+                        return true;
+                    }
+                }
+            }
+            if (testClassPredicate.isPresent()) {
+                if (testClassPredicate.get().getPredicate().test(className)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+    }
+
+    /**
+     * This tracks beans etc from the platform that have already been generated. There is no need to spend time
+     * generating them again on a hot reload
+     */
+    static class ExistingClasses {
+        Set<String> existingClasses = new HashSet<>();
+    }
+
+    private static class BytecodeTransformerConsumer implements Consumer<BytecodeTransformer> {
+
+        private final BuildProducer<BytecodeTransformerBuildItem> bytecodeTransformer;
+
+        public BytecodeTransformerConsumer(BuildProducer<BytecodeTransformerBuildItem> bytecodeTransformer) {
+            this.bytecodeTransformer = bytecodeTransformer;
+        }
+
+        @Override
+        public void accept(BytecodeTransformer t) {
+            bytecodeTransformer.produce(new BytecodeTransformerBuildItem(t.getClassToTransform(), t.getVisitorFunction()));
+        }
+    }
+}

```

Comment:
Any reason why this file is modified at all? :-)

Reply:
sorry, it seems I had added windows line feeds. I fixed that. The only changes are to add the new producer to the same BuildItem than the other logger producer.